sdk/python/kfp/compiler/_component_builder.py

```diff
@@ -29,36 +28,31 @@ class GCSHelper(object):
   """ GCSHelper manages the connection with the GCS storage """
 
   @staticmethod
-  def upload_gcs_file(local_path, gcs_path):
+  def get_blob_from_gcs_uri(gcs_path):
+    from google.cloud import storage
     pure_path = PurePath(gcs_path)
     gcs_bucket = pure_path.parts[1]
     gcs_blob = '/'.join(pure_path.parts[2:])
     client = storage.Client()
-    bucket = client.get_bucket(gcs_bucket)
+    bucket = client.bucket(gcs_bucket)
```

Comment:
We should use get_bucket, which would raise exception if the bucket does not exist.
bucket() simply creates a local object without verification.

Reply:
I agree that this function can be private but the exception raised by blob operations does not return messages that are easy to understand compared to the get_bucket.

Reply:
I tried locally with 
bucket = client.bucket(gcs_bucket)
blob = bucket.blob(gcs_blob)
blob.delete()
It would output error that are not intuitive when the bucket does not exist.

Reply:
Here is the error when I get in our hosted notebook when I use `.get_bucket` instead of `.bucket`:

    bucket.reload(client=self)
  File "/usr/local/lib/python3.5/dist-packages/google/cloud/storage/_helpers.py", line 108, in reload
    _target_object=self)
  File "/usr/local/lib/python3.5/dist-packages/google/cloud/_http.py", line 293, in api_request
    raise exceptions.from_http_response(response)
google.api_core.exceptions.Forbidden: 403 GET 
https://www.googleapis.com/storage/v1/b/ml-pipeline-playground?projection=noAcl
: 420130321805-compute@developer.gserviceaccount.com does not have storage.buckets.get access to ml-pipeline-playground.

How about we just properly check `if blob.exists():` ?

Reply:
I think get_bucket outputs very relevant messages. I still do not understand why you want to change to bucket(). Bucket() call does not involve a HTTP request, which does nothing but postpones the error. WDYT?

Reply:
The only reason is permissions.

Let me try and find a way to get good error messages.